1-Authentication/2-sign-in-b2c/SPA/src/app/app.component.ts

```diff
@@ -2,86 +2,202 @@ import { Component, OnInit, Inject, OnDestroy } from '@angular/core';
+                            account.idTokenClaims?.oid === idtoken.oid
+                            && account.idTokenClaims?.sub === idtoken.sub
+                            && ((account.idTokenClaims as IdTokenClaimsWithPolicyId).acr === b2cPolicies.names.signUpSignIn
+                                || (account.idTokenClaims as IdTokenClaimsWithPolicyId).tfp === b2cPolicies.names.signUpSignIn)
+                        );
+
+                    let signUpSignInFlowRequest: SsoSilentRequest = {
+                        authority: b2cPolicies.authorities.signUpSignIn.authority,
+                        account: originalSignInAccount
+                    };
+
+                    // silently login again with the signUpSignIn policy
+                    this.authService.ssoSilent(signUpSignInFlowRequest);
+                }
+
+                /**
+                 * Below we are checking if the user is returning from the reset password flow.
+                 * If so, we will ask the user to reauthenticate with their new password.
+                 * If you do not want this behavior and prefer your users to stay signed in instead,
+                 * you can replace the code below with the same pattern used for handling the return from
+                 * profile edit flow (see above ln. 74-92).
+                 */
+                if (idtoken.acr === b2cPolicies.names.resetPassword || idtoken.tfp === b2cPolicies.names.resetPassword) {
+                    let signUpSignInFlowRequest: RedirectRequest | PopupRequest = {
+                        authority: b2cPolicies.authorities.signUpSignIn.authority,
+                        prompt: PromptValue.LOGIN, // force user to reauthenticate with their new password
+                        scopes: []
+                    };
+
+                    this.login(signUpSignInFlowRequest);
+                }
+
+                return result;
+            });
+
+        this.msalBroadcastService.msalSubject$
+            .pipe(
+                filter((msg: EventMessage) => msg.eventType === EventType.LOGIN_FAILURE || msg.eventType === EventType.ACQUIRE_TOKEN_FAILURE),
+                takeUntil(this._destroying$)
+            )
+            .subscribe((result: EventMessage) => {
+                // Checking for the forgot password error. Learn more about B2C error codes at
+                // https://learn.microsoft.com/azure/active-directory-b2c/error-codes
+                if (result.error && result.error.message.indexOf('AADB2C90118') > -1) {
+                    let resetPasswordFlowRequest: RedirectRequest | PopupRequest = {
+                        authority: b2cPolicies.authorities.resetPassword.authority,
+                        scopes: [],
+                    };
+
+                    this.login(resetPasswordFlowRequest);
+                };
+            });
+    }
+
+    setLoginDisplay() {
+        this.loginDisplay = this.authService.instance.getAllAccounts().length > 0;
```

Comment:
Nit, magic number:

`this.loginDisplay = !!this.authService.instance.getAllAccounts().length;`